### ISSUE TITLE
How do I support solidjs? Can you implement a slate-solid？

### ISSUE THREAD

**Issue by Liangzhenquan** [ianstormtaylor/slate#5588]
I want to write an editor in solidjs and slate, but slate only has a Slate-React component, can I write a Slate-solid component based on Slate-React?

**Comment by adjourn** [comment]
Not really, JSX is where the similarities end.. 
* Rendering paradigms couldn't be more different
* Solid doesn't have synthetic event system on top of native events

You could probably build it on top of `slate-react` to reuse some of the code but be prepared to rewrite all the components and everything to do with events **but**.. I'm afraid this is the layer where bulk of the WYSIWYG editor difficulties and bugs are, I would not recommend it as a solo project.

**Comment by dylans** [comment]
There's [slate-angular](https://github.com/worktile/slate-angular) as an example of a non-React version of Slate. There's also slate-vue though it is not maintained as far as I can tell.

I wouldn't be opposed to introducing a slate-solid (or other variants) to slate itself, but we really need to add a good test suite for slate-react and anything new.

**Comment by Noobulater** [comment]
I'd contribute to a bounty to get this library working for solidjs

**Comment by bmingles** [comment]
Have you seen at this repo? https://github.com/poudels14/slate-solid

Haven’t tested but looks like someone has at least started this idea.

**Comment by Noobulater** [comment]
> Have you seen at this repo? https://github.com/poudels14/slate-solid
> 
> Haven’t tested but looks like someone has at least started this idea.

Yeah, unfortunately the repo broke somewhere along the development of solid, been a while so I don't remember which version exactly. Tried to pay a freelancer to clean it up, but their attempt was wildly unsuccessful, and I'm just wanting to throw money at the conversion rather than learn or attempt it myself.

**Comment by bmingles** [comment]
@Noobulater if you ever decide to issue a bounty I’d be interested in a link to the details.

**Comment by Noobulater** [comment]
> @Noobulater if you ever decide to issue a bounty I’d be interested in a link to the details.

If [this](https://www.slatejs.org/examples/richtext) had a functional equivalent in solid for the following, I'd easily be willing to drop $3,000 usd on it. If that sounds reasonable we can discuss the requirements in a more formal way.

**Comment by bmingles** [comment]
@Noobulater I'd be interested in talking about this more formally. My linkedin profile should be visible on my github profile if you want to connect with me there, or let me know how I can contact you elsewhere.

**Comment by bmingles** [comment]
@solid-component Checkout https://github.com/slate-solid/slate-solid
